# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/linux/hotfix-ia64-coredump-warn-vga-console.patch.ia64
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

# Selection of fixes from v7.0-rc1-epic1 tree not present in T2 6.19 ia64 patch
#
# - Fix WARN in fs/binfmt on process coredump
#   (https://github.com/linux-ia64/linux-ia64/commit/a36b6e096bd78e2f4c4269462298bb1d5b52c265)
# - Fix VGA console
#   (https://github.com/linux-ia64/linux-ia64/commit/e416d4adce3c13ba0df3124fb52ebcf5d677db3c,
#    https://github.com/linux-ia64/linux-ia64/commit/69dfd4d525bae75504279bf2789b790b1c0e7ab5)
#
# The commits were modified to work on 6.19 kernel

diff --git a/arch/ia64/kernel/ptrace.c b/arch/ia64/kernel/ptrace.c
index 3ece9a0d98ed..1eee29a9ec81 100644
--- a/arch/ia64/kernel/ptrace.c
+++ b/arch/ia64/kernel/ptrace.c
@@ -1928,14 +1928,14 @@ access_uarea(struct task_struct *child, unsigned long addr,
 
 static const struct user_regset native_regsets[] = {
 	{
-		.core_note_type = NT_PRSTATUS,
+		USER_REGSET_NOTE_TYPE(PRSTATUS),
 		.n = ELF_NGREG,
 		.size = sizeof(elf_greg_t), .align = sizeof(elf_greg_t),
 		.regset_get = gpregs_get, .set = gpregs_set,
 		.writeback = gpregs_writeback
 	},
 	{
-		.core_note_type = NT_PRFPREG,
+		USER_REGSET_NOTE_TYPE(PRFPREG),
 		.n = ELF_NFPREG,
 		.size = sizeof(elf_fpreg_t), .align = sizeof(elf_fpreg_t),
 		.regset_get = fpregs_get, .set = fpregs_set, .active = fpregs_active
diff --git a/arch/ia64/kernel/setup.c b/arch/ia64/kernel/setup.c
index 044e1772f463..2db97457ea99 100644
--- a/arch/ia64/kernel/setup.c
+++ b/arch/ia64/kernel/setup.c
@@ -609,19 +609,15 @@ setup_arch (char **cmdline_p)
 	cpu_init();	/* initialize the bootstrap CPU */
 	mmu_context_init();	/* initialize context_id bitmap */
 
-#ifdef CONFIG_VT
-	if (!conswitchp) {
-# if defined(CONFIG_VGA_CONSOLE)
-		/*
-		 * Non-legacy systems may route legacy VGA MMIO range to system
-		 * memory.  vga_con probes the MMIO hole, so memory looks like
-		 * a VGA device to it.  The EFI memory map can tell us if it's
-		 * memory so we can avoid this problem.
-		 */
-		if (efi_mem_type(0xA0000) != EFI_CONVENTIONAL_MEMORY)
-			conswitchp = &vga_con;
-# endif
-	}
+#if defined(CONFIG_VT) && defined(CONFIG_VGA_CONSOLE)
+	/*
+	 * Non-legacy systems may route legacy VGA MMIO range to system
+	 * memory.  vga_con probes the MMIO hole, so memory looks like
+	 * a VGA device to it.  The EFI memory map can tell us if it's
+	 * memory so we can avoid this problem.
+	 */
+	if (efi_mem_type(0xA0000) != EFI_CONVENTIONAL_MEMORY)
+		vgacon_register_screen(&screen_info);
 #endif
 
 	/* enable IA-64 Machine Check Abort Handling unless disabled */
@@ -642,7 +638,7 @@ setup_arch (char **cmdline_p)
 		init_smp_config();
 #endif
 
-	screen_info_setup();
+	primary_display_setup();
 	paging_init();
 
 	clear_sched_clock_stable();
